Firestore/Source/Remote/FSTDatastore.mm

```diff
@@ -299,22 +301,22 @@ - (void)invokeRPCWithFactory:(GRPCProtoCall * (^)(void))rpcFactory
                 errorHandler:(FSTVoidErrorBlock)errorHandler {
   // TODO(mikelehen): We should force a refresh if the previous RPC failed due to an expired token,
   // but I'm not sure how to detect that right now. http://b/32762461
-  [self.credentials
-      getTokenForcingRefresh:NO
-                  completion:^(FSTGetTokenResult *_Nullable result, NSError *_Nullable error) {
-                    error = [FSTDatastore firestoreErrorForError:error];
-                    [self.workerDispatchQueue dispatchAsyncAllowingSameQueue:^{
-                      if (error) {
-                        errorHandler(error);
-                      } else {
-                        GRPCProtoCall *rpc = rpcFactory();
-                        [FSTDatastore prepareHeadersForRPC:rpc
-                                                databaseID:&self.databaseInfo->database_id()
-                                                     token:result.token];
-                        [rpc start];
-                      }
-                    }];
-                  }];
+  [self.credentials getTokenForcingRefresh:NO
+                                completion:^(const Token &result, NSError *_Nullable error) {
```

Comment:
This is a bit of a semantic mess. If there was an error there should be no token. This is why in the old signature both are nullable. Only one can be non-null.

We really need something like `util::StatusOr` to represent this.

For now let's make it clearer at the call site that we're generating an invalid token. If you added something like this to `Token` it would be clearer.

```c++
static Token Token::Invalid() {
  return {"", User::Unauthenticated()};
}
```

Reply:
done